yaml/semgrep/consistency/lang-consistency-typescript.yaml

```diff
@@ -0,0 +1,20 @@
+rules:
+  - id: lang-consistency-typescript
+    patterns:
+      - pattern: $X
+      - pattern-either:
+          - pattern-inside: |
+              languages: ...
+      - metavariable-regex:
+          metavariable: $X
+          regex: ^typescript$
+    message: >-
+      Found '$X' in language config which diverges from semgrep.dev normalization.
+    languages:
+      - yaml
+    severity: WARNING
+    fix: ts
```

Comment:
Most of the other languages are spelled out - I don't care which we choose, but is it more consistent to spell this and JavaScript out as well?

Reply:
I agree with this, however semgrep.dev does not: 
https://github.com/returntocorp/semgrep-langs/blob/2cb383112a7780e5dfe4d768161c3d4ddb0bac57/lang.json#L229-L231

Reply:
Hm, gotcha 👍